### ISSUE TITLE
List components should return count of elements

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-vue-components#11]
List components (FileFormats, ServiceTypes, UdfRuntimes, ...) should return count of elements similar to how SupportedFeatures has it. Ensure that this works properly with the reactivity.

**Comment by m-mohr** [comment]
Available with release 1.0.0-beta.3.

**Comment by christophfriedrich** [comment]
About the idea of using a computed property instead of methods: I now tried that with the service types in the Hub.

Added to SupportedServiceTypes.vue in openeo-vue-components:
```js
	computed: {
		count: function() {
			return this.serviceTypeNames.length;
		}
	},
```

Added to Backend.vue in openeo-hub:
```js
    computed: {
        supportedServiceTypesCount() {
            //if(this.$refs.supportedServiceTypesComponent != undefined)
            return this.$refs.supportedServiceTypesComponent.count;
        },
```

Result: Type error because when the computed property function is called, the SupportedServiceTypes component isn't ready yet. Thinking about it, this isn't really a surprise.

When wrapping that line with the `return` into an if that checks whether the ref is not `undefined`, nothing happens at all because the computed property function is only called once (when the ref is still `undefined`), but not again when the ref is finally available.

~I doubt there's something like a watcher on a ref.~
I actually checked and it appears to be possible: https://stackoverflow.com/questions/39035498/vuejs-watch-refs But in the same thread there's also an answer by Linus Borg (a major Vue guy) who says it won't work (i.e. not intended), so maybe even if it _is_ possible, it may not be the _elegant_ way.

But all these ideas would be a very hacky way to access things anyway.

The only "right" way in Vue to provide data from a child component back to its parent is via events. So once ready, the SupportedServiceTypes component could emit a `countAvailable` event and the parent component could listen for such an event and then write the transmitted value to a local data item which can be reactively used in the template. That would probably be 3-4 lines of code.

What do you think? Leave as-is, go the hacky way or set up events?

**Comment by m-mohr** [comment]
tl;dr: Thanks for the experiments. Events seem to be strange for this kind of information. I think we can leave it as is.

Long version: In principle we should remove getCount functions altogether, decouple logic from view and then export the logic to be used by other components, but for such small things like a count I'm fine with leaving this. In theory the hub could count, too. For all parts except the features this is pretty straightforward (data.length) and for the features the logic is mostly in featurelist.js already. Events seem to be over the top for a simple count. It would probably lead to more code than otherwise written for the "workarounds" (counting / getCount).
